server/container_create.go

```diff
@@ -495,26 +495,49 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID string,
 		capabilities := linux.GetSecurityContext().GetCapabilities()
 		toCAPPrefixed := func(cap string) string {
 			if !strings.HasPrefix(strings.ToLower(cap), "cap_") {
-				return "CAP_" + cap
+				return "CAP_" + strings.ToUpper(cap)
 			}
 			return cap
 		}
+
+		// Add/drop all capabilities if "all" is specified, so that
+		// following individual add/drop could still work. E.g.
+		// AddCapabilities: []string{"ALL"}, DropCapabilities: []string{"CHOWN"}
+		// will be all capabilities without `CAP_CHOWN`.
+		// see https://github.com/kubernetes/kubernetes/issues/51980
+		if inStringSlice(capabilities.GetAddCapabilities(), "ALL") {
+			for _, c := range getOCICapabilitiesList() {
+				if err := specgen.AddProcessCapability(c); err != nil {
+					return nil, err
+				}
+			}
+		}
+		if inStringSlice(capabilities.GetDropCapabilities(), "ALL") {
+			for _, c := range getOCICapabilitiesList() {
+				if err := specgen.DropProcessCapability(c); err != nil {
+					return nil, err
+				}
+			}
+		}
+
 		if capabilities != nil {
 			addCaps := capabilities.AddCapabilities
-			if addCaps != nil {
-				for _, cap := range addCaps {
-					if err := specgen.AddProcessCapability(toCAPPrefixed(cap)); err != nil {
-						return nil, err
-					}
+			for _, cap := range addCaps {
+				if strings.ToUpper(cap) == "ALL" {
+					continue
```

Comment:
This could be a break; since theoretically all caps were already added, although it would prevent reporting of failures if the caller specified a bad capability.
Probably best to just leave it alone.

Reply:
Yeah, let's leave it the way it is now

Reply:
Not sure this is documented anywhere, but we might want to mention it.

If you specify Add ALL or Drop All, the Add all happens first then the drop all, then individual add followed by individual drops.

That way you can say

DROPALL add sys_admin and only have sys_admin
Or 
ADDALL but drop sys_admin and have all without SYS_ADMIN

Reply:
this is how docker behaves afaict, where do you want to document it?